src/generators/java/presets/JacksonPreset.ts

```diff
@@ -19,5 +19,43 @@ export const JAVA_JACKSON_PRESET: JavaPreset = {
       }
       return renderer.renderBlock([content]);
     },
+  },
+  enum: {
+    self({renderer, content}) {
+      renderer.addDependency('import com.fasterxml.jackson.annotation.*;');
+      return content;
+    },
+    additionalContent({ renderer, model }) {
+      const enumName = renderer.nameType(model.$id);
+      const type = Array.isArray(model.type) ? 'Object' : model.type;
+      const classType = renderer.toClassType(renderer.toJavaType(type, model));
+
+      return `private ${classType} value;
+    
+${enumName}(${classType} value) {
+  this.value = value;
+}
+    
+@JsonValue
+public ${classType} getValue() {
+  return value;
+}
+
+@Override
+public String toString() {
+  return String.valueOf(value);
+}
+
+@JsonCreator
+public static ${enumName} fromValue(${classType} value) {
+  for (${enumName} e : ${enumName}.values()) {
+    if (e.value.equals(value)) {
+      return e;
+    }
+  }
+  throw new IllegalArgumentException("Unexpected value '" + value + "'");
+}`;
+    },
```

Comment:
I see we are kinda missing two preset hooks, one that renders the `fromValue` and `getValue` methods. 

Because this way you can do something like:

```ts
    getValue({content}) {
      return `@JsonValue
${content}`
    },
    fromValue({content}) {
      return `@JsonValue
${content}`;
    },
```

Cause the problem with the current implementation is that it completely overwrites whatever the enum renderer generates which should rarely be the case for presets.

Do you want to take a crack at adding those preset hooks @micro-jumbo ?

Reply:
@jonaslagoni added new presets: `getValue`, `fromValue`. Additionally added `ctor` preset as having constructor in `additionalContent` looked a bit ackward

Reply:
Awesome 👍